File: contract-examples/test/warp.ts
// (c) 2019-2022, Ava Labs, Inc. All rights reserved.
// See the file LICENSE for licensing terms.

import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
import { expect } from "chai";
import {
  Contract,
  ContractFactory,
} from "ethers"
import { ethers } from "hardhat"

const fundedAddress: string = "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
const WARP_ADDRESS = "0x0200000000000000000000000000000000000005";

File: tests/precompile/solidity/warp.go
// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
// See the file LICENSE for licensing terms.

// Implements solidity tests.
package solidity

import (
	"context"

	"github.com/ava-labs/avalanchego/api/health"
	"github.com/ava-labs/subnet-evm/tests/utils"
	ginkgo "github.com/onsi/ginkgo/v2"
	"github.com/onsi/gomega"
)

var _ = ginkgo.Describe("[Warp]", ginkgo.Ordered, func() {
	ginkgo.It("Setup Subnets", ginkgo.Label("Warp"), func() {
		client := health.NewClient(utils.DefaultLocalNodeURI)
		healthy, err := client.Readiness(context.Background())
		gomega.Expect(err).Should(gomega.BeNil())
		gomega.Expect(healthy.Healthy).Should(gomega.BeTrue())
	})

	// Steps:
	// Set up two subnets with the same genesis
	// send a message from blockchainA to blockchainB
	// verify message on blockchainB from blockchainA
})
